packages/overlay/src/vaadin-overlay-mixin.js

```diff
@@ -292,6 +292,13 @@ export const OverlayMixin = (superClass) =>
     /** @private */
     _openedChanged(opened, wasOpened) {
       if (opened) {
+        // Prevent possible errors on re-opening when updating a property on the
+        // already disconnected owner element causes `opened` to be set to `true`
+        if (this.owner && this.getRootNode().host === this.owner && !this.owner.isConnected) {
```

Comment:
Would it be good enough to check if the overlay itself is disconnected?

That also makes me wonder if we have a problem in general here. Was it previously possible to set a dialog for example to be `opened` and only afterwards attach it to the DOM?

Reply:
> Was it previously possible to set a dialog for example to be opened and only afterwards attach it to the DOM?

Looks like that works at the moment.

Reply:
Yes, the initial update is performed by LitElement when the element is first connected to the DOM.

> Would it be good enough to check if the overlay itself is disconnected?

Thanks, that makes sense. Updated the PR and moved the test to basic suite.